openedx/core/djangolib/model_mixins.py

```diff
@@ -12,3 +12,23 @@ def __init__(self, *args, **kwargs):  # pylint: disable=unused-argument
         Override to kill usage of this model.
         """
         raise TypeError("This model has been deprecated and should not be used.")
+
+
+class RetireUserByEmailMixin:
+    """
+    Mixin allows inheriting models to delete users by their email field. A 
+    Jenkins job that removes user data calls a model's "retire_user".
+    """
+    def __init__(self):
+        assert hasattr(self, 'email'), 'Inheriting model must have an "email" field.'
```

Comment:
You heart is in the right place with this, but I don't think this will really work as intended.  I'd suggest moving this assertion into the `retire_user()` method.